core/src/main/java/org/sonarsource/sonarlint/core/issue/IssueService.java

```diff
@@ -97,17 +89,15 @@ public IssueService(ConfigurationRepository configurationRepository, ServerApiPr
     this.telemetryService = telemetryService;
   }
 
-  public void changeStatus(ChangeIssueStatusParams params, CancelChecker cancelChecker) {
-    var configurationScopeId = params.getConfigurationScopeId();
+  public void changeStatus(String configurationScopeId, String issueKey, ResolutionStatus newStatus, boolean isTaintIssue, CancelChecker cancelChecker) {
```

Comment:
At some point should we introduce an intermediate type if there are a lot of parameters?

Reply:
This is hiding the dust under the carpet :) A method with many parameters is maybe a sign of bad design. In this example, I expect at some point to have an issue UUID and to not have to tell if this is a taint or not.